src/d3a/models/strategy/external_strategies/storage.py

```diff
@@ -262,19 +272,23 @@ def _bid(self, payload):
     def _bid_impl(self, arguments, bid_response_channel):
         try:
             assert arguments["energy"] <= self.state.energy_to_buy_dict[self.next_market.time_slot]
+
+            replace_existing = arguments.get('replace_existing', True)
```

Comment:
IMO you can skip this, as you already asserted for all arguments in line 256.

Reply:
The assertion on line `256` doesn't check whether all arguments are in the request, but rather whether all the arguments that are in the request are allowed. This is something that changed compared to the previous behavior.

`replace_existing` is not mandatory, so we should allow requests that do not contain it. In that case, the default value I set is `True`.

Reply:
yes, sorry, I didnt understand tha, but left another comment there https://github.com/gridsingularity/d3a/pull/1038/files/761256609509ec4fc92a817b14c9bf2cbcb9bcba#r569274825